.gitlab-ci.yml

```diff
@@ -23,17 +22,23 @@ cache:
 
 before_script:
   - python -V  # Print out python version for debugging
-  - pip install -r requirements.txt
-  - pip install python-coveralls
+  - pip install -r requirements.txt python-coveralls pylint pycodestyle
 
 test:
   stage: test
   script:
-    - echo $TEST_SUITE
-    - echo $TEST_SUITE > test.sh
-    - bash test.sh
-    - coverage report
-  #  - coveralls
+    # TODO bjuhasz: use tox because "setup.py test" is deprecated:
+    # https://setuptools.readthedocs.io/en/latest/setuptools.html#test-build-package-and-run-a-unittest-suite
+    - coverage run --branch --source=pyega3 -m pytest
+    # The $COVERAGE_FAIL_UNDER and $PYLINT_FAIL_UNDER variables are defined in GitLab:
+    # https://gitlab.ebi.ac.uk/EGA/ega-download-client/-/settings/ci_cd , under Variables.
+    # TODO bjuhasz: increase the values of $COVERAGE_FAIL_UNDER and $PYLINT_FAIL_UNDER
```

Comment:
When we reach 80-90% coverage, we should increase the current values of the $COVERAGE_FAIL_UNDER and $PYLINT_FAIL_UNDER variables in GitLab to 80-90%, so that we never fall below 80-90% again.